src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -116,6 +131,7 @@ def get(self, svc_id):
         return {
             'osd_map': self.get_osd_map(svc_id),
             'osd_metadata': mgr.get_metadata('osd', svc_id),
+            'orchestrator': self._get_orch_state(int(svc_id), self.get_removing_osds())
```

Comment:
What about something like:
```suggestion
            'operational_status': ('deleting`|`creating`|`working`)
```

I don't think `unknown` is a good status here: if we cannot know the actual operational status of an OSD, then we have an issue... Alternatively we can say `other` but I think we should be able to state that an OSD is in a known/precise status...

BTW, from a RESTful perspective, this should work this way (but I think we have more RESTless stuff than this):
```
# once a DELETE is issued, further requests 
DELETE /osd/15
# async delete
200 (OK) / 202 (Accepted) / 302 (Moved temp) -> Location: /task/<id>

GET /osd/15
404 (Not Found) / 302 (Moved Temp) -> Location: /task/<id>

# and the status should be retrieved from the asynchronous task
GET /task/<id>
200
```

Reply:
> I don't think unknown is a good status here: if we cannot know the actual operational status of an OSD, then we have an issue

Any suggestion for OSDs not managed by the orchestrator? just `working`?

Reply:
Understood: so this is the same issue as with hosts info, coming from 2 different sources, right? Well, in that case, what about something like `unmamanged`?

Reply:
Thanks, `unmanaged` sounds great. I refactored the PR and use it.

> BTW, from a RESTful perspective, this should work this way (but I think we have more RESTless stuff than this):

After some thinking, I'll prefer exposing the status in OSD resource responses. The reason is that removing state tracked by the orchestrator is kept in persistent storage: removing operation might take a long time, if the mgr daemon is terminated and restarted, the state will still be there (comparing to the thread-manager we currently use in the Dashboard, which will be re-initialized).